client/homebrew/editor/editor.jsx

```diff
@@ -151,7 +151,7 @@ const Editor = createClass({
 
 						// definition lists
 						if(line.includes('::')){
-							const regex = /^([^\n]*?)::([^\n]*)(?:\n|$)/ym;
+							const regex = /^([^\n]*?\s)::(\s[^\n]*)(?:\n|$)/ym;
```

Comment:
This regex requires the `::` to have white space on each side, and so something like `term::defn` won't match.

What you want is to not match any line that starts with a colon.
`/^([^:][^\n]*?)::([^\n]*)(?:\n|$)/ym;`

Reply:
Is it the worst thing to require spaces on each side? To me that is just cleaner. I could do both too

Reply:
It would be inconsistent with what is recognised by the parser as a DT/DD, which means the `::` won't get highlighted even when it is successfully parsed as a DT/DD. The highlighting there is particularly helpful since the text just runs along, so the lack of highlighting is egregious.

Hmm .. suggested regex still isn't right, since `:term :: defn` is parsed as a dt/dd. The thing we don't want to match is a run of colons with no other text.

Reply:
Okay fair enough.  I have the regex now set so that it doesn't require spaces on either side of the colons.  It also checks for a colon at the start of a line (and prevents highlighting if one exists).  The regex can be seen below, and I just pushed the commit.

Another thing in this commit:  I changed how the "range" used in highlighting is derived, though in this commit I only changed one line and before going ahead with a merge I could apply the same change to other similar "range checks" for highlighting.  

Below is the code with my change, what was just committed, though you can see in the other two lines the original method.

```js
						// definition lists
						if(line.includes('::')){
							const regex = /^([^\n:]+?)::([^\n]*)(?:\n|$)/ymd;
							let match;
							while ((match = regex.exec(line)) != null){
								codeMirror.markText({ line: lineNumber, ch: line.indexOf(match[0]) }, { line: lineNumber, ch: line.indexOf(match[0]) + match[0].length }, { className: 'define' });
								codeMirror.markText({ line: lineNumber, ch: line.indexOf(match[1]) }, { line: lineNumber, ch: line.indexOf(match[1]) + match[1].length }, { className: 'term' });
								codeMirror.markText({ line: lineNumber, ch: match.indices[2][0] }, { line: lineNumber, ch: match.indices[2][1] }, { className: 'definition' });
							}
						}
```

I noticed that there was some odd highlighting behavior, and how it checked where the `span`s were applied in the editor (the spans differentiate between "term" and "definition" and apply "define" to everything *not* included in the other spans).  This was evident when your line was just `cat dog :: ` and no definition typed yet.  It would take the ` ` (space) at the end as the "definition" match (`match[2]`) and then when we want to actually apply the class name for the highlight, it would find the ` ` between `cat dog` because we just did `indexOf` on the string again.  It would apply `definition` class to that first space.

With my change, I used the `d` flag in the regex, which makes the `exec()` command return the indices of the match groups back as an array of arrays.  I then just use those indices to set the range, or bounds, of the `span`s.  

It's likely this same technique could be applied to the Injection syntax highlighting, and inline spans?  Though I'd be satisfied to just apply it to definition lists for now.

That `d` tag was neat, I didn't know about it until now.  It may be helpful with all the variables/reflinks talk.